src/_pytest/nodes.py

```diff
@@ -265,12 +256,20 @@ def setup(self) -> None:
     def teardown(self) -> None:
         pass
 
-    def listchain(self) -> List["Node"]:
-        """Return list of all parent collectors up to self, starting from
-        the root of collection tree.
+    def iterchain(self) -> Iterator["Node"]:
+        """Iterate over self and all parent collectors up to root of the
+        collection tree, starting from self.
 
-        :returns: The nodes.
+        .. versionadded:: 8.1
         """
+        parent: Optional[Node] = self
+        while parent is not None:
+            yield parent
+            parent = parent.parent
+
+    def listchain(self) -> List["Node"]:
+        """Return a list of all parent collectors up to and including self,
+        starting from the root of the collection tree."""
         chain = []
```

Comment:
while at it maybe also add 
```suggestion
        chain = [*self.iterchain()]
        chain.reverse()
```

Reply:
On my machine (Python 3.11), for a length 7 chain, the existing implementation takes ~0.4 microseconds and reusing iterchain takes ~0.7 microseconds.

I vaguely recall `listchain` showing up in profiles so I decided to keep the faster implementation. If you think it's not worth it, I can change it as well.

Reply:
I see, in that case let's skip until we get better profiles

Thanks for the detail